src/test/protobuf/test.proto

```diff
@@ -36,6 +36,10 @@ message MyTest {
   }
 
   map<fixed64, bytes> fixed64_to_bytes = 18;
+
+  optional uint32 uint = 19;
```

Comment:
can you port this test: https://github.com/google/protobuf/blob/master/java/util/src/test/java/com/google/protobuf/util/JsonFormatTest.java#L349 to make sure all range checks are done correctly?

Reply:
Done, and I also added explicit tests for sint* and fixed* types. Thanks for the careful review!